Added damage interfaces

BrokenKage opened the pull request:
Added two interfaces. IDamage which holds methods for changing Living Entity general attributes. IStrengthsWeakness which holds methods for Living Entity mobs strengths and weaknesses. 

The purpose of this was to step strength, speed, haste, etc. effects away from potion effects into a more flexible interface.

BrokenKage committed:
Added damage interfaces

Parsonswy commented on src/net/peacefulcraft/sco/swordskills/utilities/IStrengthWeakness.java:
```diff
@@ -0,0 +1,13 @@
+ * outgoing and incoming damage modifiers.
+ */
+public interface IStrengthWeakness {
+    List<String> getDamageModifiers();
```
Is this meant to be a list of things that have `.set...` things in `IDamage`? Further, what is the source of those modifiers? I am assuming a combination of an Entities' base abilities, Items, and SwordSkills, but wanted to clarify.

BrokenKage committed:
Added modifier class.

Converted string damage modifiers to modifier class. Removed weakness and strength modifier lists for one list. Modifiers contain boolean to determine if it is a weakness or strength.

Changed damage modifiers on Active Mobs to initialize as copy of mythicmob instance. Allows for dynamic changing via swordskills.
